java/core/src/java/org/apache/orc/impl/reader/tree/StructBatchReader.java

```diff
@@ -20,11 +20,18 @@
 import org.apache.hadoop.hive.ql.exec.vector.ColumnVector;
 import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
 import org.apache.orc.impl.OrcFilterContextImpl;
+import org.apache.orc.impl.PositionProvider;
 import org.apache.orc.impl.TreeReaderFactory;
+import org.apache.orc.impl.reader.StripePlanner;
 
 import java.io.IOException;
 import java.util.Set;
 
+/**
+ * Handles the Struct rootType for batch handling. The handling assumes that the root
+ * {@link org.apache.orc.impl.TreeReaderFactory.StructTreeReader} has no nulls, this is required as
+ * the {@link VectorizedRowBatch} does not represent the root Struct as a vector.
```

Comment:
Nice, I would rephrase to something like: ..has no nulls. Root Struct vector is not represented as part of the final  the {@link  org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch}.

Reply:
Changed